tests/core/test_validator.py

```diff
@@ -40,6 +36,33 @@
     assert validator.verify_utterances()
 
 
+async def test_verify_story_structure():
```

Comment:
let's use `pytest.mark.parametrize` for this and `test_verify_bad_story_structure_ignore_warnings`